sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1724,12 +1724,18 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
 void DispatchNativeKernel(void *Blob) {
   // First value is a pointer to Corresponding CGExecKernel object.
   CGExecKernel *HostTask = *(CGExecKernel **)Blob;
+  bool MayDeleteCG = static_cast<bool>(*((void **)Blob + 1));
 
   // Other value are pointer to the buffers.
-  void **NextArg = (void **)Blob + 1;
+  void **NextArg = (void **)Blob + 2;
   for (detail::Requirement *Req : HostTask->MRequirements)
     Req->MData = *(NextArg++);
   HostTask->MHostKernel->call(HostTask->MNDRDesc, nullptr);
+
+  // The command group will (if not already was) be released in scheduler.
+  // Hence we're free to deallocate it here.
+  if (MayDeleteCG)
```

Comment:
I am curious about the naming of `MayDeleteCG`. I cannot not see the `May` uncertainty here.

Reply:
Nice catch. I've changed the name.